Added the close icon button

gokul112 opened the pull request:
<!-- Please read the contribution guide before contributing https://github.com/shahednasser/sbuttons/blob/master/CONTRIBUTING.md -->
<!-- Please describe what changes or additions this pull request pertain to -->
Added a New close icon button

status  | Button name    | Description      | Submitted by |
|--------|----------------|------------------|--------------|
| '\f00d';| ~~**Close Button**~~| A button that is used to close/cancel windows. |[gokul](https://github.com/gokul112) | 

<!-- Specify the issue it relates to, if any --->
Issue:
https://github.com/shahednasser/sbuttons/issues/506

![Screenshot 2020-10-02 at 3 24 38 PM](https://user-images.githubusercontent.com/11469509/94913115-528dea00-04c6-11eb-8c3b-751bf18f469f.png)

HADES-01 commented on src/components/icon/_close.less:
```diff
@@ -0,0 +1,22 @@
+.close-btn {
+    &:before {
```
It should be `&::before` not `&:before`

shahednasser commented on assets/js/buttons-examples.js:
```diff
@@ -199,6 +199,12 @@ $(document).ready(function () {
+          label: "Close Button",
+          classes: "close-btn",
+          text: "",
+        }
```
Please change this object to the following:

```
        {
          type: "close",
          label: "Close Button",
          classes: "",
          variations: [
             {
                 classes: "close-btn",
                 label: ""
             }
          ]
        }
```

This will make the close button appear without the button colors, just on its own.

shahednasser replied:
Also place this button under Icon in the correct alphabetical order

shahednasser commented on src/components/icon/_close.less:
```diff
@@ -0,0 +1,22 @@
+      padding: 0 0.5em;
+    }
+  
+    &.rounded-btn {
```
Remove this rule as it won't be necessary anymore

shahednasser commented on src/components/icon/_close.less:
```diff
@@ -0,0 +1,22 @@
+      -webkit-font-smoothing: antialiased;
+    }
+  
+    &:not(.rounded-btn) {
```
remove this rule as it won't be necessary anymore.

shahednasser reviewed (changes requested):
Please make sure to compile the code once you're done by running `npm run compile` to see the changes.

gokul112 commented:
Addressed the comments.Please review it 
![Screenshot 2020-10-02 at 9 07 34 PM](https://user-images.githubusercontent.com/11469509/94942027-5be17b80-04f3-11eb-8e84-87513b7bb199.png)

HADES-01 reviewed (commented):
I don't think that you have ran the `npm run compile command` as there is still no change in the dist/sButtons.css file.

gokul112 commented:
> I don't think that you have ran the `npm run compile command` as there is still no change in the dist/sButtons.css file.

Hi @HADES-01 
I ran the npm run compile but no need to push the dist file am I right?

HADES-01 reviewed (approved):
My bad, Its fine now.

shahednasser reviewed (changes requested):
Thank you for your work @gokul112 . There's actually something we need to work out first in buttons-examples before you can make this button look better, but for now I need you to run compile and commit the changes. I can't see what you have done with the button.

gokul112 commented:
> Thank you for your work @gokul112 . There's actually something we need to work out first in buttons-examples before you can make this button look better, but for now I need you to run compile and commit the changes. I can't see what you have done with the button.

Hi @shahednasser,
As i understand from your comment you want me to push the dist files, is that required?
I Have attached the screenshot for this please check it.

shahednasser commented:
Can you please add shouldHaveRoundedType and shouldHaveBlockType to your object in `buttons-examples.js` and set them to false? Also remove the border of the close icon.

@gokul112 I can see the icon now but there is a problem with merging from master, as some buttons are not working anymore.

Can you please get your repo up to date with this repo or re-fork this repo, then create a new branch with your changes? 

I'm closing this PR awaiting a new PR!
